compiler/rustc_typeck/src/collect.rs

```diff
@@ -2889,17 +2889,7 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
                     .emit();
                     InlineAttr::None
                 } else if list_contains_name(&items[..], sym::always) {
-                    if tcx.sess.instrument_coverage() {
-                        // Fixes Issue #82875. Forced inlining allows LLVM to discard functions
-                        // marked with `#[inline(always)]`, which can break coverage reporting if
-                        // that function was referenced from a coverage map.
-                        //
-                        // FIXME(#83429): Is there a better place, e.g., in codegen, to check and
-                        // convert `Always` to `Hint`?
-                        InlineAttr::Hint
-                    } else {
-                        InlineAttr::Always
-                    }
+                    InlineAttr::Always
```

Comment:
Oh, before you make this change, we need to test that this doesn't break the brotli_decompressor

See #82875

I was not able to create a coverage test that failed in the way it fails in this issue, but by not forcing inline always, the issue was resolved.

Reply:
I just tested the example from that issue. It fails on my main rustc (2021-03-24 nightly) but succeeds on my new branch. So the fix for inline always works.

Reply:
That's great. 

Before merging, I'd like to try your patch on some creates I had issues with.

Reply:
I cherry-picked your branch commits and am testing them now.